M2/Macaulay2/packages/VirtualResolutions.m2

```diff
@@ -137,46 +137,30 @@ resolveViaFatPoint(Ideal, Ideal, List) := ChainComplex => (J, irr, A) -> (
 -- Note: the Determinatal strategy is based on Theorem 1.3 of [Loper2019].
 --------------------------------------------------------------------
 --------------------------------------------------------------------
-isVirtual = method(Options => {Strategy => null})
-isVirtual (Ideal, ChainComplex) := Boolean => opts -> (irr, C) -> (
+isVirtual = method(TypicalValue => Boolean, Options => {Strategy => null})
+isVirtual(NormalToricVariety, ChainComplex) := opts -> (X,   C) -> isVirtual(ideal X, C)
+isVirtual(Ideal,              ChainComplex) := opts -> (irr, C) -> (
+    S := ring irr;
+    if S =!= ring C then error "isVirtual: expected objects in the same ring";
 -- if strategy "determinantal is selected, the method checks virtuality
 -- via the depth criterion on the saturated ideals of minors
+    debugInfo := if debugLevel < 1 then identity else i -> printerr("isVirtual failed at homological degree ", toString i);
     if opts.Strategy === "Determinantal" then (
         for i from 1 to length(C) do (
             if rank(source(C.dd_i)) != (rank(C.dd_i) + rank(C.dd_(i+1))) then (
-                if debugLevel >= 1 then print "isVirtual failed at homological degree " | toString i;
-                return false;
-                );
-            );
+                debugInfo i; return false));
         for i from 1 to length(C) do (
             minor := minors(rank(C.dd_i),C.dd_i);
             minorSat := ourSaturation(minor,irr);
-            if depth(minorSat,ring(minorSat)) < i then (
-                if debugLevel >= 1 then print "isVirtual failed at homological degree " | toString i;
```

Comment:
here, too

Reply:
> here, too

This is also old code, by the way.

Reply:
Do you mean that it is never entered?  If so, we could get rid of it.

Reply:
I don't understand what you mean. You left this review directly on the old code, and so there's nothing I can do about this. This segment of the new code has been majorly changed. Have you seen the new code?

Reply:
Oh, now I see what you mean!

Reply:
Great! Any other suggestions on this pull request?